### DISCUSSION TITLE
Help getting a Raspberry Pi CANBUS hat going

### DISCUSSION THREAD

**Discussion by NGTOne** [kairos-io/kairos#2517] [Q&A]
I'm trying to get a [PiCAN2 Duo](https://copperhilltech.com/pican2-duo-can-bus-board-for-raspberry-pi/) going on a RasPi 4B with Ubuntu 22.04+Kairos, but nothing I do seems to make it work - it never detects the two CAN ports the board exposes. I've built an image with the settings from the PiCAN2 documentation in both `/boot/config.txt` and `/boot/firmware/config.txt`, and neither seems to have any effect.

```
$ cat /boot/config.txt 
# https://www.raspberrypi.com/documentation/computers/config_txt.html
# https://emteria.com/kb/connect-uart-rpi
# https://www.raspberrypi.com/documentation/computers/config_txt.html
enable_uart=1
uart_2ndstage=1

# Disabling bluetooth might change the serial device:
# https://raspberrypi.stackexchange.com/a/69721
# On rpi4, if bluetooth is not disabled (like here), the device is /dev/ttyS0
#dtoverlay=disable-bt
dtparam=i2c_arm=on
# dtparam=i2c_arm_baudrate=400000
dtparam=spi=on
dtoverlay=mcp2515-can0,oscillator=16000000,interrupt=25
dtoverlay=mcp2515-can1,oscillator=16000000,interrupt=24
dtoverlay=spi-bcm2835-overlay
```

```
$ sudo /sbin/ip link set can0 up type can bitrate 500000
Cannot find device "can0"
```

I've verified it works on a clean Raspbian, so I know it's something in the Kairos/Ubuntu configuration.

**Comment by sutyum** [comment]
Wonder if your issue was eventually resolved. I am trying to assess Kairos for a medical device which uses UART

**Reply by NGTOne** [reply]
Nope, never got it going. Project timeline forced me to switch back to Raspbian and I've not had time to experiment with it since.

**Comment by jimmykarily** [comment]
Created a ticket from this discussion to investigate: https://github.com/kairos-io/kairos/issues/3036
